storage/src/adb/benches/variable_init.rs

```diff
@@ -41,111 +46,182 @@ cfg_if::cfg_if! {
-            let v = vec![(rng.next_u32() % 255) as u8; ((rng.next_u32() % 24) + 20) as usize];
-            db.update(rand_key, v).await.unwrap();
-            if rng.next_u32() % COMMIT_FREQUENCY == 0 {
-                db.commit(None).await.unwrap();
-            }
-        }
-        db.commit(None).await.unwrap();
-        db.prune(db.inactivity_floor_loc()).await.unwrap();
-        db.close().await.unwrap();
+fn gen_random_store(cfg: Config, num_elements: u64, num_operations: u64) {
+    let runner = Runner::new(cfg.clone());
+    runner.start(|ctx| async move {
+        let store_cfg = store_cfg();
+        let db = Store::init(ctx, store_cfg).await.unwrap();
+        gen_random_kv(db, num_elements, num_operations).await;
     });
 }
 
+/// Generate a large key-value db with random data. The function seeds the db with exactly
+/// `num_elements` elements by inserting them in order, each with a new random value. Then, it
+/// performs `num_operations` over these elements, each selected uniformly at random for each
+/// operation. The ratio of updates to deletes is configured with `DELETE_FREQUENCY`. The database
+/// is committed after every `COMMIT_FREQUENCY` operations.
+async fn gen_random_kv<A: Db<Context, <Sha256 as Hasher>::Digest, Vec<u8>, EightCap>>(
+    mut db: A,
+    num_elements: u64,
+    num_operations: u64,
+) {
+    // Insert a random value for every possible element into the db.
+    let mut rng = StdRng::seed_from_u64(42);
+
+    for i in 0u64..num_elements {
+        let k = Sha256::hash(&i.to_be_bytes());
+        // Generate a random value with a length between 24 and 40 bytes (avg = 32).
+        let v = vec![(rng.next_u32() % 255) as u8; ((rng.next_u32() % 16) + 24) as usize];
+        db.update(k, v).await.unwrap();
+    }
+
+    // Randomly update / delete them.
+    for _ in 0u64..num_operations {
+        let rand_key = Sha256::hash(&(rng.next_u64() % num_elements).to_be_bytes());
+        if rng.next_u32() % DELETE_FREQUENCY == 0 {
+            db.delete(rand_key).await.unwrap();
+            continue;
+        }
+        // Generate a random value with a length between 20 and 44 bytes (avg = 32).
+        let v = vec![(rng.next_u32() % 255) as u8; ((rng.next_u32() % 24) + 20) as usize];
+        db.update(rand_key, v).await.unwrap();
+        if rng.next_u32() % COMMIT_FREQUENCY == 0 {
+            db.commit().await.unwrap();
+        }
+    }
+    db.commit().await.unwrap();
+    db.prune(db.inactivity_floor_loc()).await.unwrap();
+    db.close().await.unwrap();
+}
+
 type AnyDb = Any<Context, <Sha256 as Hasher>::Digest, Vec<u8>, Sha256, EightCap>;
 
+type StoreDb = Store<Context, <Sha256 as Hasher>::Digest, Vec<u8>, EightCap>;
+
+#[derive(Debug, Clone, Copy)]
+enum Variant {
+    Unauthenticated,
+    Any,
+}
+
+impl Variant {
+    pub fn name(&self) -> &'static str {
+        match self {
+            Variant::Unauthenticated => "unauthenticated",
```

Comment:
nit: prefer calling this `store` unless we change its name to `unauthenticated`?

Reply:
changed to store to avoid increasing diff too much and give us time to contemplate...